test/backend_test.in.cpp

```diff
@@ -1382,6 +1382,66 @@ NGRAPH_TEST(${BACKEND_NAME}, convert_int32_float32)
     EXPECT_EQ((vector<float>{1, 2, 3, 4}), read_vector<float>(result));
 }
 
+NGRAPH_TEST(${BACKEND_NAME}, convert_float64_int32)
+{
+    Shape shape{2, 3};
+    auto A = make_shared<op::Parameter>(element::f64, shape);
+    auto f = make_shared<Function>(make_shared<op::Convert>(A, element::i32), ParameterVector{A});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    // Create some tensors for input/output
+    auto a = backend->create_tensor(element::f64, shape);
+    copy_data(a,
+              vector<double>{std::numeric_limits<double>::min(),
+                             std::numeric_limits<double>::max(),
+                             std::numeric_limits<double>::lowest(),
+                             INFINITY,
+                             -INFINITY,
+                             4.0});
+    auto result = backend->create_tensor(element::i32, shape);
+
+    auto handle = backend->compile(f);
+    backend->call_with_validate(handle, {result}, {a});
+    EXPECT_EQ((vector<int>{0,
+                           std::numeric_limits<int>::lowest(),
```

Comment:
If I c++ static_cast a max double into an int I get the answer 2147483647 which is std::numeric_limits<int>::max(). 
So I think the results should be:
```
0
std::numeric_limits<int>::max()
std::numeric_limits<int>::lowest()
std::numeric_limits<int>::max()
std::numeric_limits<int>::lowest()
4
```
Same comment for the float test below.
Although I really don't know what the correct answer is for the INFINITY

Reply:
Which version of the compiler you used?
All these things are not defined well by the compiler standard. 
I just want to provide a backend behavior as it CPU backend does (this is expected by the user).
Perhaps it is impossible due to diffrent compilers provide diffrent results.

Reply:
This was with gcc 7.3 which is the default on ubuntu 18
I can try again with clang

Reply:
I only tested because converting a large positive float/double to an in should be a positive result, not a negative result.

Reply:
floating point arithmetic are in the grey area of diffrent compilers.

Reply:
"I only tested because converting a large positive float/double to an int should be a positive result, not a negative result"
This is not quite right. The maximum in double datatype can't represented in int datatype. There is no representations in this case. The CPU instructions used for this return "error status" but the exception is masked by the CPU. So, If we ask to convert "double::maximum" to int - it will be seen as "int::minimum".

Reply:
This is what the C99 spec has to say on the issue:

`When a finite value of real floating type is converted to an integer type other than _Bool, the fractional part is discarded (i.e., the value is truncated toward zero). If the value of the integral part cannot be represented by the integer type, the behavior is undefined.`

https://stackoverflow.com/questions/526070/handling-overflow-when-casting-doubles-to-integers-in-c

Reply:
@rkimballn1 "If the value of the integral part cannot be represented by the integer type, the behavior is undefined."
This is exactly part that I told.
But the question is - what we should do with it? I think the user would expect the operation behavior exactly as it done by CPU backend. OpenCL (the background compiler under INTELGPU) has it's own understanding of these things.
This is why I tried to make this PR.

Reply:
I think the user shouldn't be expecting any consistent behavior when they drift outside of what the language definition; they should fix their model to stay within the defined behavior.
However, our users are actually python users, so python behavior, not C/C++ behavior, is what we should be emulating. What does Python do here?

Reply:
Ok, I changed this PR to be INTELGPU specific.
We need these changes to pass, at least, TF test system using the backend.